packages/app/src/cli/services/deploy/upload.ts

```diff
@@ -207,45 +207,42 @@ async function uploadFunctionExtension(
   await compileFunctionExtension(extension, options, url)
 
   const query = api.graphql.AppFunctionSetMutation
-  const schemaVersions = Object.values(extension.metadata.schemaVersions).shift()
-  const schemaMajorVersion = schemaVersions?.major
-  const schemaMinorVersion = schemaVersions?.minor
-
   const variables: api.graphql.AppFunctionSetVariables = {
-    uuid: options.identifier,
-    extensionPointName: getFunctionExtensionPointName(extension.configuration.type),
     title: extension.configuration.name,
     description: extension.configuration.description,
-    force: true,
-    schemaMajorVersion: schemaMajorVersion === undefined ? '' : `${schemaMajorVersion}`,
-    schemaMinorVersion: schemaMinorVersion === undefined ? '' : `${schemaMinorVersion}`,
-    configurationUi: extension.configuration.configurationUi,
-    moduleUploadUrl: url,
+    apiType: getFunctionExtensionPointName(extension.configuration.type),
     apiVersion: extension.configuration.apiVersion,
-    skipCompilationJob: true,
+    inputQuery,
     appBridge: extension.configuration.ui?.paths
       ? {
           detailsPath: extension.configuration.ui.paths.details,
           createPath: extension.configuration.ui.paths.create,
         }
       : undefined,
-    inputQuery,
+    moduleId: parseFunctionExtensionUploadURL(url),
+  }
+
+  // NOTE: This is a shim to support CLI projects that currently use the UUID instead of the ULID
```

Comment:
Yeah, nothing wrong about the comment itself. The API could be more expressive about the nature of this uuid though. Could also be through a field description 🤷.

Reply:
Yea true, I suppose I was keeping it consistent with the type/model, since it's not `legacyUuid` in that either and seems like a pain to change everywhere now. Maybe it doesn't matter if the argument isn't consistent, though.